### DISCUSSION TITLE
Can not read cookies

### DISCUSSION THREAD

**Discussion by mattholy** [fastapi/fastapi#6613] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from typing import Optional
from fastapi import FastAPI, Response, Cookie

app = FastAPI()
@app.get("/set")
async def create_cookie(response: Response):
    response.set_cookie(key="fakesession", value="fake-cookie-session-value")
    return {"message": "Come to the dark side, we have cookies"}

@app.get("/read")
async def read_items(ads_id: Optional[str] = Cookie(None)):
    return {"ads_id": ads_id}

if __name__ == "__main__":
    uvicorn.run(app, port=80)
```

### Description

When access /read ,I got `{"ads_id": null}`

### Operating System

Windows

### Operating System Details

Winodws 11

### FastAPI Version

0.70.0

### Python Version

Python 3.9.7

**Comment by libra146** [comment]
@mattholy I also encountered this problem,  can you provide some help, thank you

**Comment by mattholy** [comment, marked as answer]
> @mattholy I also encountered this problem, can you provide some help, thank you

Your variable's name must be the same as your cookie's key
e.g. in the example I provided above
if you would like to read `fakesession` in cookies
you have to make your code be like :
```python
async def read_items(fakesession: Optional[str] = Cookie(None)):
    return {"whatever": fakesession}
```

**Comment by mattholy** [comment]
@libra146 Additionally if your cookie's key is something like `X-Token` (with a dash in it), you need to read it as `x_token`

**Comment by libra146** [comment]
> @libra146 Additionally if your cookie's key is something like `X-Token` (with a dash in it), you need to read it as `x_token`

It worked, I spent a long time here, thank you for your work.

**Comment by cruncher12** [comment]
Thanks! I couldn't find anything in the FastApi docs so this was quite helpful...
